apps/server/src/shared/domain/entity/task.entity.spec.ts

```diff
@@ -28,6 +37,495 @@ describe('Task Entity', () => {
+				const spy = jest.spyOn(course, 'getSubstitutionTeacherIds');
+
+				task.createTeacherStatusForUser(user.id);
+
+				expect(spy).toHaveBeenCalled();
+			});
+
+			it('should return true if userId is part of it.', () => {
+				const user = userFactory.build();
+				user.id = '0123456789ab';
+				const course = courseFactory.build();
+				course.substitutionTeachers.add(user);
+				const task = taskFactory.build({ course });
+
+				const result = task.createTeacherStatusForUser(user.id);
+
+				expect(result.isSubstitutionTeacher).toBe(true);
+			});
+
+			it('should return false if userId not is part of it', () => {
+				const user = userFactory.build();
+				user.id = '0123456789ab';
+				const course = courseFactory.build();
+
+				const task = taskFactory.build({ course });
+
+				const result = task.createTeacherStatusForUser(user.id);
+
+				expect(result.isSubstitutionTeacher).toBe(false);
+			});
+		});
+	});
+
+	describe('isSubmittedForUser', () => {
+		it('should call getNumberOfSubmittedUsers and return true if userId is part of it.', () => {
+			const student = userFactory.build();
+			student.id = '0123456789ab';
+			const submission = submissionFactory.build({ student });
+			const task = taskFactory.build();
+			task.submissions.add(submission);
+
+			const spy = jest.spyOn(task, 'getSubmittedUserIds');
+
+			const result = task.isSubmittedForUser(student.id);
+
+			expect(spy).toHaveBeenCalled();
+			expect(result).toBe(true);
+
+			spy.mockReset();
+		});
+
+		it('should call getNumberOfSubmittedUsers and return false if userId is not part of it.', () => {
+			const student = userFactory.build();
+			student.id = '0123456789ab';
+			const submission = submissionFactory.build({ student });
+			const task = taskFactory.build();
+			task.submissions.add(submission);
+
+			const spy = jest.spyOn(task, 'getSubmittedUserIds');
+
+			const result = task.isSubmittedForUser('1');
+
+			expect(spy).toHaveBeenCalled();
+			expect(result).toBe(false);
+
+			spy.mockReset();
+		});
+	});
+
+	describe('isGradedForUser', () => {
+		it('should call getGradedUserIds and return true if userId is part of it.', () => {
+			const student = userFactory.build();
+			student.id = '0123456789ab';
+			const submission = submissionFactory.graded().build({ student });
+			const task = taskFactory.build();
+			task.submissions.add(submission);
+
+			const spy = jest.spyOn(task, 'getGradedUserIds');
+
+			const result = task.isGradedForUser(student.id);
+
+			expect(spy).toHaveBeenCalled();
+			expect(result).toBe(true);
+
+			spy.mockReset();
+		});
+
+		it('should call getGradedUserIds and return false if userId is not part of it.', () => {
+			const student = userFactory.build();
+			student.id = '0123456789ab';
+			const submission = submissionFactory.graded().build({ student });
+			const task = taskFactory.build();
+			task.submissions.add(submission);
+
+			const spy = jest.spyOn(task, 'getGradedUserIds');
```

Comment:
is it really necessary to check whether this method was called? it seems more like an implementation detail to me.

Reply:
It is more cleaner to say method must be pass a, b, c, d, e. But otherwise if c, d, e already test and is prepared only passiv by calling internally. It increase our test a lot if i write checks c, d, e twice from different views but with same logic interally. I think about it and came to not well solution which way is better.

The call is much shorter, but is danger if someone remove or changed it without realsing why the test exist.